pkg/controller/infrastructure/dualstack/dualstack.go

```diff
@@ -0,0 +1,112 @@
+package dualstack
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/gardener/gardener-extension-provider-alicloud/pkg/alicloud"
+	alicloudclient "github.com/gardener/gardener-extension-provider-alicloud/pkg/alicloud/client"
+)
+
+// DualStack contains values of EIP used to render terraform charts
+type DualStack struct {
+	Enabled            bool
+	Zone_A             string
+	Zone_A_CIDR        string
+	Zone_A_IPV6_SUBNET int
+	Zone_B             string
+	Zone_B_CIDR        string
+	Zone_B_IPV6_SUBNET int
+}
+
+// CreateDualStackValues create DualStack values
+func CreateDualStackValues(
+	enableDualStack bool,
+	region string,
+	vpcCidr *string,
+	credentials *alicloud.Credentials,
+	clientFactory alicloudclient.ClientFactory,
+) (*DualStack, error) {
+	dualStack := DualStack{
+		Enabled: enableDualStack,
+	}
+	if !enableDualStack {
+		return &dualStack, nil
+	}
+	if vpcCidr == nil {
+		return nil, fmt.Errorf("vpcCidr must be set")
+	}
+	dualStack.Zone_A_IPV6_SUBNET = 255
+	dualStack.Zone_B_IPV6_SUBNET = 254
+	nlbClient, err := clientFactory.NewNLBClient(region, credentials.AccessKeyID, credentials.AccessKeySecret)
+	if err != nil {
+		return nil, err
+	}
+	zones, err := nlbClient.GetNLBAvailableZones(region)
+	if err != nil {
+		return nil, err
+	}
+	if len(zones) < 2 {
+		return nil, fmt.Errorf("not enough available zones for DualStack")
+	}
+	dualStack.Zone_A = zones[0]
+	dualStack.Zone_B = zones[1]
```

Comment:
What is the impact that we simply use the first two zone? Will this cause imbalance in case there are more zones, e.g. the first two zones of a highly-available three zone setup will be more important because they contain the vswitches? What about clusters that only use a single zone?

Reply:
The two zones are used to deploy NLB, not for user's cluster. If user enable dual stack, a NLB need be created.